src/install/dracut-install.c

```diff
@@ -625,9 +625,14 @@ static int resolve_deps(const char *src)
         }
 
         fclose(fptr);
-        wait(&err);
-
-        return ret;
+        int err = 126;
+        while (waitpid(ldd_pid, &err, 0) == -1 && errno == EINTR)
+                ;
+        err = WIFSIGNALED(err) ? 128 + WTERMSIG(err) : WEXITSTATUS(err);
+        if (err != 0)
+                log_error("ERROR: %s %s failed with %d", ldd, fullsrcpath, err);
+
+        return ret ?: err;
```

Comment:
Let's prioritize the exit code, in other words `err ?: ret;`. This was b2d3f8610b3d5a3d385164f5459d0c3938831d11 behaviour.

If you do not mind a couple more lines of code, I would avoid those useless `err = 126;`/`ret = 126;` in the last two commits and simpy return early:

```c
        int err;
        while (waitpid(ldd_pid, &err, 0) == -1) {
                if (errno != EINTR) {
                        log_error("ERROR: waitpid() failed: %m");
                        return 1;
                }
        }
```

Reply:
Applied